tests/test_routing_lievre_tutorial.py

```diff
@@ -234,3 +236,235 @@ def test_lievre_tutorial(self):
+          :Flush             RAVEN_DEFAULT            PONDED_WATER     SURFACE_WATER         # Moves water from PONDED_WATER to SURFACE_WATER (routed to outlet). Use DEFAULT RAVEN_DEFAULT method.
+        :EndHydrologicProcesses
+        """
+
+        streaminputs = xr.open_dataset(vic_streaminputs_nc_path)
+
+        start = streaminputs.indexes["time"][0]
+        end = streaminputs.indexes["time"][-4]  # to match the tutorial end date
+
+        model.config.rvi.start_date = start.to_pydatetime()
+        model.config.rvi.end_date = end.to_pydatetime()
+
+        # Raven will use 24h even though the NC inputs are 6h
+        model.config.rvi.time_step = "24:00:00"
+
+        model.config.rvi.evaluation_metrics = [
+            "NASH_SUTCLIFFE",
+            "PCT_BIAS",
+            "KLING_GUPTA",
+        ]
+
+        #######
+        # RVH #
+        #######
+
+        rvh_extractor = RoutingProductShapefileExtractor(
+            routing_product_shp_path,
+            hru_aspect_convention="ArcGIS",
+            routing_product_version="2.1",
+        )
+        rvh_config = rvh_extractor.extract()
+        channel_profiles = rvh_config.pop("channel_profiles")
+
+        # There are 3 gauged subbasins in the HRU file generated for the Routing Product V2.1,
+        # but we are only interested in the most downstream one, "02LE024".
+
+        gauged_sbs = [sb for sb in rvh_config["subbasins"] if sb.gauged]
+        assert len(gauged_sbs) == 3
+
+        gauged_sb = None  # we're looking for gauge 02LE024 which is the most downstream
+        for sb in gauged_sbs:
+            if sb.gauge_id == "02LE024":
+                gauged_sb = sb
+            else:
+                # Raven will ignore the other two
+                sb.gauged = False
+                sb.gauge_id = ""
+
+        rvh_config.update(
+            {
+                "land_subbasin_property_multiplier": SBGroupPropertyMultiplierCommand(
+                    "Land", "MANNINGS_N", 1.0
+                ),
+                "lake_subbasin_property_multiplier": SBGroupPropertyMultiplierCommand(
+                    "Lakes", "RESERVOIR_CREST_WIDTH", 1.0
+                ),
+            }
+        )
+
+        for k, v in rvh_config.items():
+            model.config.rvh.update(k, v)
+
+        #######
+        # RVP #
+        #######
+
+        # The labels used for the following commands ("Lake_Soil_Lake_HRU", "Soil_Land_HRU", etc)
+        # must correspond to the values of certain fields of the Routing Product:
+        # LAND_USE_C, VEG_C, SOIL_PROF
+
+        model.config.rvp.tmpl = """
+        {soil_classes}
+
+        {soil_profiles}
+
+        {vegetation_classes}
+
+        {land_use_classes}
+
+        {avg_annual_runoff}
+
+        {channel_profiles}
+        """
+
+        model.config.rvp.avg_annual_runoff = 594
```

Comment:
I am happy with this being hard-coded here. But we had a function to derive those values, right?

Reply:
We do (`get_average_annual_runoff`) and I updated the two Lievre tests so that they now make use of it.

Reply:
Nice!!